File: wallet.py
class Wallet:
    def __init__(self, balance):
        self.balance = balance
        self.cards = []

    def display_balance(self):
        print(f"Solde actuel : {self.balance}")

    def add_money(self, amount):
        self.balance += amount
        print(f"Montant ajouté. Nouveau solde : {self.balance}")

    def withdraw_money(self, amount):
        if amount > self.balance:
            print("Fonds insuffisants.")
        else:
            self.balance -= amount
            print(f"Montant retiré. Nouveau solde : {self.balance}")

    def add_card(self, card):
        self.cards.append(card)
        print("Carte ajoutée :", card.get_name())

    def remove_card(self, card_name):
        card_to_remove = next((card for card in self.cards if card.get_name() == card_name), None)
        if card_to_remove:
            self.cards.remove(card_to_remove)
            print("Carte retirée :", card_to_remove.get_name())
        else:
            print("Carte non trouvée :", card_name)
    def display_all_cards(self):
        print("Cartes dans le portefeuille :")
        for card in self.cards:
            print(f"Nom : {card.get_name()}, Type : {card.get_type()}")        

File: wallet.go
package main

import "fmt"

type Wallet struct {
	Balance  float64
	Cards []*Card
}

func NewWallet(balance float64) *Wallet {
	return &Wallet{
		Balance:  balance,
		Cards: []*Card{},
	}
}

func (w *Wallet) DisplayBalance() {
	fmt.Printf("Solde actuel : %.2f\n", w.Balance)
}

func (w *Wallet) AddMoney(amount float64) {
	w.Balance += amount
	fmt.Printf("Montant ajouté. Nouveau solde : %.2f\n", w.Balance)
}

func (w *Wallet) WithdrawMoney(amount float64) {
	if amount > w.Balance {
		fmt.Println("Fonds insuffisants.")
	} else {
		w.Balance -= amount
		fmt.Printf("Montant retiré. Nouveau solde : %.2f\n", w.Balance)
	}
}

func (w *Wallet) AddCard(card *Card) {
	w.Cards = append(w.Cards, card)
	fmt.Printf("Carte ajoutée : %s\n", card.GetName())
}

func (w *Wallet) RemoveCard(cardName string) {
	for i, card := range w.Cards {
		if card.GetName() == cardName {
			w.Cards = append(w.Cards[:i], w.Cards[i+1:]...)
			fmt.Printf("Carte retirée : %s\n", card.GetName())
			return
		}
	}
	fmt.Printf("Carte non trouvée : %s\n", cardName)
}

func (w *Wallet) DisplayAllCards() {
	fmt.Println("Cartes dans le portefeuille :")
	for _, card := range w.Cards {
		fmt.Printf("Nom : %s, Type : %s\n", card.GetName(), card.GetType())
	}
}
